fineract-provider/src/test/java/org/apache/fineract/cob/loan/CheckLoanRepaymentOverdueBusinessStepTest.java

```diff
@@ -95,11 +94,12 @@ public void givenLoanWithNoInstallmentOverdueAfterConfiguredDaysWhenStepExecutio
         // given
         when(configurationDomainService.retrieveRepaymentOverdueDays()).thenReturn(1L);
         LocalDate loanInstallmentRepaymentDueDateBefore5Days = DateUtils.getBusinessLocalDate().minusDays(5);
-        Collection<OverdueLoanScheduleData> overdueInstallmentsForLoan = Arrays
-                .asList(new OverdueLoanScheduleData(1L, 1L, loanInstallmentRepaymentDueDateBefore5Days.toString(), BigDecimal.valueOf(0.0),
-                        "", "", BigDecimal.valueOf(0.0), BigDecimal.valueOf(0.0), 1));
-        when(loanReadPlatformService.retrieveAllOverdueInstallmentsForLoan(any())).thenReturn(overdueInstallmentsForLoan);
         Loan loanForProcessing = Mockito.mock(Loan.class);
+        List<LoanRepaymentScheduleInstallment> loanRepaymentScheduleInstallments = Arrays
+                .asList(new LoanRepaymentScheduleInstallment(loanForProcessing, 1, LocalDate.now(ZoneId.systemDefault()),
+                        loanInstallmentRepaymentDueDateBefore5Days, BigDecimal.valueOf(0.0), BigDecimal.valueOf(0.0),
+                        BigDecimal.valueOf(0.0), BigDecimal.valueOf(0.0), false, new HashSet<>(), BigDecimal.valueOf(0.0)));
```

Comment:
Would you please extend your test suite with the following:
numberOfDaysAfterDueDateToRaiseEvent = 1
There are 2 installments
- First is fully paid
- Second is not fully paid

Run the job on the first installment due date. Outcome should be nothing. (no event raised)
Run the job on the first installment due date + 1. Outcome should be nothing. (no event raised) 
Run the job on the first installment due date + 2. Outcome should be nothing (no event raised)

Run the job on the second installment due date. Outcome should be nothing. (no event raised)
Run the job on the second installment due date + 1. Outcome: event raised! 
Run the job on the second installment due date + 2. Outcome should be nothing (no event raised)